libraries/chain/worker_evaluators.cpp

```diff
@@ -88,6 +88,16 @@ namespace golos { namespace chain {
             logic_exception::cannot_create_techspec_for_premade_worker_proposal,
             "Cannot create techspec for premade worker proposal");
 
+        if (wpo->type == worker_proposal_type::premade_work) {
+            GOLOS_CHECK_LOGIC(o.worker == wpo_post.author,
```

Comment:
Why worker?
The author of proposal should be the same as the author of tspec.

Worker is an developer, who can implement task.